src/main/java/com/pinterest/secor/common/SecorConfig.java

```diff
@@ -276,6 +276,8 @@ public boolean getBoolean(String name, boolean defaultValue) {
         return mProperties.getBoolean(name, defaultValue);
     }
 
+    public boolean getUploadOffsetFile() { return getBoolean("secor.upload_offset_file", false); }
```

Comment:
Probably the method name should be called 'useUploadOffsetFile', the parameter name probably should be called 'use_upload...' to indicate it's boolean type.